android/KMEA/app/src/main/java/com/tavultesoft/kmea/packages/PackageProcessor.java

```diff
@@ -70,7 +70,9 @@ public PackageProcessor(File resourceRoot) {
    * @throws IllegalArgumentException
    */
   public String getPackageID(File path) {
-    String filename = path.getName().toLowerCase();
+    // Sometimes, downloading a kmp multiple times results in a filename with a suffix:
+    // " (#).kmp" or "-#.kmp" in the filename, so strip out the number
+    String filename = path.getName().toLowerCase().replaceAll("(\\s+\\(|-)\\d+\\)?.kmp$", ".kmp");
```

Comment:
I guess this is fine but technically it'd match `khmer_angkor-10).kmp` as well 😆 Also `.kmp` should really be `\\.kmp`, otherwise it matches `xkmp` etc.

You could use the regex `/((\s+\(\d+\))|(-\d+))\.kmp$/` (omitted Java string escapes for readability here!)

Reply:
ok - I tweaked the regex again (and pulled the spaces out of the groupings)